src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs

```diff
@@ -148,6 +150,84 @@ public WorkspaceInfo()
+                double cy = 0;
+                double zoom = 1.0;
+                double scaleFactor = 1.0;
+                string id = "";
+                string category = "";
+                string description = "";
+                string version = "";
+                var runType = RunType.Manual;
+                int runPeriod = RunSettings.DefaultRunPeriod;
+                bool hasRunWithoutCrash = false;
+                bool isVisibleInDynamoLibrary = true;
+
+                cx = double.Parse(jObject["View"]["X"].ToString(), CultureInfo.InvariantCulture);
+                cy = double.Parse(jObject["View"]["Y"].ToString(), CultureInfo.InvariantCulture);
+                zoom = double.Parse(jObject["View"]["Zoom"].ToString(), CultureInfo.InvariantCulture);
+                scaleFactor = double.Parse(jObject["View"]["Dynamo"]["ScaleFactor"].ToString(), CultureInfo.InvariantCulture);
+                funName = jObject["Name"].ToString();
+                id = jObject["Uuid"].ToString();
+                category = jObject["Category"].ToString();
+                description = jObject["Description"].ToString();
+                hasRunWithoutCrash = bool.Parse(jObject["View"]["Dynamo"]["HasRunWithoutCrash"].ToString());
+                isVisibleInDynamoLibrary = bool.Parse(jObject["View"]["Dynamo"]["IsVisibleInDynamoLibrary"].ToString());
+                version = jObject["View"]["Dynamo"]["Version"].ToString();
+                if (forceManualExecutionMode || !Enum.TryParse(jObject["View"]["Dynamo"]["RunType"].ToString(), false, out runType))
+                {
+                    runType = RunType.Manual;
+                }
+                runPeriod = Int32.Parse(jObject["View"]["Dynamo"]["RunPeriod"].ToString());
+
+                // we have a dyf and it lacks an ID field, we need to assign it
+                // a deterministic guid based on its name.  By doing it deterministically,
+                // files remain compatible
+                if (string.IsNullOrEmpty(id) && !string.IsNullOrEmpty(funName) && funName != "Home")
+                {
+                    id = GuidUtility.Create(GuidUtility.UrlNamespace, funName).ToString();
+                }
+
+                workspaceInfo = new WorkspaceInfo
+                {
+                    ID = id,
+                    Name = funName,
+                    X = cx,
+                    Y = cy,
+                    Zoom = zoom,
+                    ScaleFactor = scaleFactor,
+                    FileName = path,
+                    Category = category,
+                    Description = description,
+                    Version = version,
+                    RunType = runType,
+                    RunPeriod = runPeriod,
+                    HasRunWithoutCrash = hasRunWithoutCrash,
+                    IsVisibleInDynamoLibrary = isVisibleInDynamoLibrary
+                };
+                return true;
+            }
+            catch (Exception ex)
+            {
```

Comment:
so in the case that one of those many properties is missing from the workspace info section of the Dynamo block, do we fail to open the workspace? One potentially nice thing about JSON format is that humans can create it fairly easily, would be good if they can open their hand crafted graphs without adding all this dynamo specific stuff.

Reply:
also, this seems like it would stop us from loading any graphs that do not have a view block, or do we do the right thing if we fail to create a workspace info?

Reply:
Not any graphs, dyf only. It is fragile but I do not think workspace info constructor is that robust either. Yes, dyf would not open properly if infomation are missing, similar to XMLs. I feel like I do not know enough about which properties are essential for customNodeworkspace initialization yet..

Reply:
@mjkkirschner Re-wrote the deserialize logic, should be more robust now